### PR TITLE
More Crates And Some Changes To Other Crates

### PR DESCRIPTION
**Pull request by Isy232**
## About The Pull Request

Adds more crates to cargo with the goal of giving other departments useful crates to purchase. Also makes some crates a lot less overpriced, adds bulk crates for some items. Should go well with #1607.

Changes:
- Advanced engineering tools crate, RCD, Jaws of life, mostly the stuff on the CE's belt for 2000.
- Advanced surgical tools for 1500.
- Department hardsuits can now be purchased for 8000.
- Atmospherics hardsuits for 24000.
- Jetpacks can be bought for 2000.
- Hardsuit jetpack upgrades can be bought for 2000.
- Materials crates. (Fairly pricey, miners are not being replaced).
Edit: Bluespace crystals are not purchasable, they are the only exempt material.
- Bulk options for existing and new crates.
- Chef crates have been made cheaper.
- Interesting floor tile crates have been made cheaper.
- Chemical implant crates have been made cheaper.
- Blood crates made cheaper.
- IV crates made cheaper.
- Defibrillator crates made cheaper.
- RPED crates made cheaper.
- Floor tiles crates made cheaper.
- Miner starter kit made cheaper.
- Basic cargo supplies crate made cheaper.

I'm sure that's all the additions, perhaps not all the price changes. A lot of these crates were pretty much scams.

Generally making a lot of crates cheaper, all weapons crates are untouched though, no crates have been removed however some have been moved in order or had their contents changed.

No crates have been removed, some have been re-ordered to make more sense.

Also adds stacks for various materials, it was needed for the crate changes, this will not affect in game at all.

## Why It's Good For The Game

As of now, most departments do not really have any reason to buy anything from cargo unless your creating a new engine, need more guns or mindshields. This PR aims to add more cargo crates and re-balance existing ones so departments may actually have use for their budget cards.

## Changelog
:cl:
add: Adds new cargo crates.
tweak: Some cargo crates cost less or have more items.
/:cl:

### CODE REVIEW

**Comment by Cenrus** [comment]
Big powercreep, makes the crew far too strong. Alien organs are basically free alien tech every game, and making them expensive is meaningless as budget cards generate millions with no effort required from cargo.

Purchasable bluespace crystals and anomaly cores take the importance away from mining, so we'll have Phazons running around way more often.

Autodocs are supposed to be late game tech because they subtract from medical doctor jobs, making them available at cargo is a bad idea.

**Comment by Isy232** [comment]
>Big powercreep, makes the crew far too strong. Alien organs are basically free alien tech every game, and making them expensive is meaningless as budget cards generate millions with no effort required from cargo.

Alien organs cannot be used to get alien tech, at least none of the ones I have made purchasable.

Budget cards got nerfed, the science one did at least and that was the only one I know of that was capable of generating millions.

>Purchasable bluespace crystals and anomaly cores take the importance away from mining, so we'll have Phazons running around way more often.

I have purposefully made bluespace crystals the only material not purchasable, shoulda wrote that down on the inital post, added it now.

I did add the stack code so it is possible to add bluespace polycrystal crates in the future, but they are not available currently.

Mining rarely ever brings any anomaly cores back to the station, I've had it happen maybe twice in 400 hours of playtime on science. More often your stuck waiting and hoping that the RNG gods spawn an anomaly somewhere on the station for you to neutralize. This PR adds another way to get anomaly cores, however I think its even more difficult than waiting for an anomaly. However it is not reliant upon RNG, which is nice.

I know you can get them from delaminating the SM, but that's risky and still has its use as it can mass produce anomaly cores.

>Autodocs are supposed to be late game tech because they subtract from medical doctor jobs, making them available at cargo is a bad idea.

Autodocs can only implant organs into people. This means they can only perform surgeries that put things into people. This only gives you the machine board without the parts, you still have to order or make those separately and construct the machine. If science wanted to, they could research autodocs within 20 minutes its just people rarely do because it costs almost an absurd amount of research points for these and robotics likely wants mechs and people want parts. Also, an autodoc with low tier parts takes 2 or so minutes per surgery, making doing the surgery with the tools still have a purpose even if an autodoc is available.

I could also remove this crate from the PR or make it more expensive if people still remain unconvinced.

If a lot of people hate on a certain crate, I'll drop it from this PR and perhaps try it later. Goal here is to make cargo actually have useful things to buy for departments, which is severely lacked before.

**Comment by VictorVow** [comment]
- Advanced engi tools, surgical tools, autodoc, rapid syringe gun - no, that's what research is for both lorewise and gameplaywise
- Department hardsuits - huge powercreep, these would be bought up massively for being spaceproof, flashproof and having extra resistances
- Anomaly core crate - money for this is lame, I'm probably going to port the anomaly core via Toxins stuff that tg has instead
- Jetpack and hardsuit jetpack upgrade - too cheap, with multiple departments easily getting >$100k in late game this should be more like 6k and 10k
- Alien organs - just no

**Comment by Cenrus** [comment]
Neutralizing anomalies is a scientists job. Autodocs usually result in people getting loaded with cybernetics too, which is why I think adding them to cargo is a bad idea.

**Comment by Isy232** [comment]
>Autodocs usually result in people getting loaded with cybernetics too, which is why I think adding them to cargo is a bad idea.

Cybernetics still need to be researched, there is no crate for those.

>Advanced engi tools, surgical tools, autodoc - no, that's what research is for both lorewise and gameplaywise

CE spawns with a toolbelt with these same tools on them, so lorewise it is obvious that NT knows how to make advanced tools, just that they are expensive. Gameplaywise cargo is utterly useless most of the time. The only thing of value to order for engineering is maybe more metal or insulated gloves if you actually run out. Advanced tools are a fairly early tech and do not cost many points at all and then in comparison to the materials used to make them these crates are quite the scam. If you want these tools just a bit quicker you can order them from cargo.

> Anomaly core crate - money for this is lame, I'm probably going to port the anomaly core via Toxins stuff that tg has instead

Anomalies are shitty RNG spawn, but since victor agrees and may port another way of generating them I'll drop them from this PR.

>rapid syringe gun

Rapid Syringe gun crate moved to normal syringe guns.

>Alien organs - just no

Dropping alien organs crate from this PR, I still think its a decent idea just not worth arguing here.

>Department hardsuits - huge powercreep, these would be bought up massively for being spaceproof, flashproof and having extra resistances

These have a large amount of slowdown making them a bit of a pain to use. I do not think they are massively useful for the same reason that people do not immediately rush to the hardsuits at round start to wear them.
If you want flashproof, get a pair of welding goggles, welding helmet or crack a security locker for a pair of sunglasses.
If you want extra resistances, a radsuit or armour vest does it equally as well or better with less slowdown while being more available and cheaper.
Spaceproof can be accomplished with simple EVA gear thats cheaper and much more common.
If you want spaceproof armour, SWAT crates cover that even better.
The hardsuits offer nothing special other than being a combination of a bunch of things useful to a department.

>Jetpack and hardsuit jetpack upgrade - too cheap, with multiple departments easily getting $100k in late game this should be more like 6k and 10k

You need a space suit or hardsuit to use a jetpack in the first place, so that adds on to the cost of a jetpack.

I do not know of any departments that can get >$100k in anything less than a 2 hour round

**Comment by Cenrus** [comment]
The downside of cybernetics is that they take forever to install using surgery and someone else has to do it for you. Just let science have their tech instead of giving it to cargo for a payment.

**Comment by Isy232** [comment]
> The downside of cybernetics is that they take forever to install using surgery and someone else has to do it for you. Just let science have their tech instead of giving it to cargo for a payment.

Alright, dropping autodoc crate.

**Comment by VictorVow** [comment]
> I do not think they are massively useful for the same reason that people do not immediately rush to the hardsuits at round start to wear them.

On MRP they don't get rushed much due to powergaming rules but you'd be lucky to find an engi or atmos hardsuit on LRP, and it's because they're powerful. 

> The hardsuits offer nothing special other than being a combination of a bunch of things useful to a department.

That's the whole point, you can get these things if you pick up a several different items but the unique advantage is that you can get it all in one item. Mmm yes, fireproof and spaceproof from atmos hardsuit, sign me up! Complete immunity from any atmos antag shenanigans all because budget cards make stupid amounts of money out of thin air. There's a reason there's only one of these on the station.

**Comment by Isy232** [comment]
> That's the whole point, you can get these things if you pick up a several different items but the unique advantage is that you can get it all in one item. Mmm yes, fireproof and spaceproof from atmos hardsuit, sign me up! Complete immunity from any atmos antag shenanigans all because budget cards make stupid amounts of money out of thin air. There's a reason there's only one of these on the station.

I was thinking this too, hence I made it cost more, I can remove atmos suit from purchase or make it far more expensive, whatever you think. Looking back I think the number is too low for the atmos suit now. 

There are already 3 atmos immune suits on the station, HoS suit is 95% resistant and many others have high resistances.
Then the 2 atmospherics firesuits.
More atmos suits are already obtainable through xenobiology requiring a fairly high tier slime.

I think they could have a price, but $6,000 is far too cheap, something like $10,000-16,000 would seem better to me, or just not make them purchasable for previous mentioned reasons. I feel like the CE should be able to get a single additional atmospherics hardsuit if they spend most of their money.

Another thing I am thinking is if atmospherics hardsuits do become too easy to get, it could easily lead to really stupid strategies for cultists, such as plasmaflood the summoning room and have cultists wearing atmospherics hardsuits. Same for gangs, holofan on top of the dominator beacon in a plasmaflood.

Also, mechs are mostly atmospherics proof, requiring pretty much fusion temperatures to damage if the wiki is correct.

> On MRP they don't get rushed much due to powergaming rules but you'd be lucky to find an engi or atmos hardsuit on LRP, and it's because they're powerful.

I'm partially assuming that is due to people hoarding the suits to avoid not having one when shit hits the fan.
Also, newer players taking the suits because they are not sure what they are doing, but the hardsuit looks cool and its supposedly important to their job.
Most experienced players who I recognize barely wear hardsuits unless they need it for their job or the station is fucked..

$4,000 is a bit cheap, I'm more thinking $6,000 to $8,000.
Perhaps medical ones a little cheaper since base medical ones are not flash immune, still $6,000 at minimum. (CMO's hardsuit is a special subtype of medical hardsuit with flash immunity).

Anyways, whats your opinion or opinions?

**Comment by VictorVow** [comment]
> There are already 3 atmos immune suits on the station, 

1 hardsuit which has full atmos immunity, 2 firesuits which are not spaceproof. The firesuits don't give cold temperature immunity but even they are pretty strong.

> HoS suit is 95% resistant and many others have high resistances.

High resistance is not immunity, you still won't be able to walk the hallways during a plasmaflood. There's many emergency firefighter suits as well, so I don't get your argument that all this exists so suddenly full immunity should too. 

> Anyways, whats your opinion or opinions?

Hardsuits should be very expensive (especially atmos hardsuit) if available because they'd become the most popular items behind insuls and guns immediately.

**Comment by Isy232** [comment]
> Hardsuits should be very expensive (especially atmos hardsuit) if available because they'd become the most popular items behind insuls and guns immediately.

Can you give me a number, I'm thinking 12,000 per atmos suit and $8000 per all other hardsuits, exception of medical being $6000 since it lacks flash protection.

**Comment by VictorVow** [comment]
I'd say 24k for atmos hardsuit to limit people abusing it with plasmaflooding, other hardsuit costs are fine.

**Comment by Isy232** [comment]
How about 12k and if it gets abused then I will ensure to increase it, while I do think team antags abusing plasma fires to win all fights is a possibility, even with xenobiology able to make tons of atmos proof suits its still a strategy I've never seen used intentionally.

24k is a ton, I sat for 30 minutes and the engi budget only got up to $20k while all other budgets were at most half that to the point where all together the total value wasn't much at all,

Looking at economy code, it looks like all the budgets combined get 35k at shift start or something divided between all of the different budgets.

Economy Code: `var/fuck = alive_crew / crew`

**Comment by VictorVow** [comment]
Science and Medbay get the most from my recollection, seen $200k+ many a time, you're underestimating how quickly they go up.

**Comment by Isy232** [comment]
Looking at the code it looks like it goes up depending on some factors that I cant read. I'll set to 24k and if they never get bought or are just too unaffordable I can change it then.

File: code/modules/cargo/packs.dm [hunk 1]
```diff
@@ -346,6 +344,20 @@
...
+	desc = "Contains a security hardsuit for catching criminals in space! Requires Security access to open."
+	cost = 8000
+	contains = list(/obj/item/clothing/suit/space/hardsuit/security)
+	crate_name = "security hardsuit crate"
+
+/datum/supply_pack/security/hardsuit3
+	name = "Bulk Security Hardsuit Crate"
+	desc = "Contains three security hardsuits for catching criminals in space! Requires Security access to open."
+	cost = 24000
+	contains = list(/obj/item/clothing/suit/space/hardsuit/security)
```
Comment (PowerfulBacon) [inline code/modules/cargo/packs.dm]: Is this only 1 suit?

Comment (Isy232) [reply code/modules/cargo/packs.dm]: Heck, your right, I'll fix that

**Comment by Isy232** [comment]
Riiight, issues should be fixed now.

**Pull request merged**